src/providers/create/components/EsxiProviderEdit.tsx

```diff
@@ -0,0 +1,100 @@
+import { type FC, useCallback, useState } from 'react';
+import { validateEsxiURL } from 'src/modules/Providers/utils/validators/provider/vsphere/validateEsxiURL';
+import { validateVDDKImage } from 'src/modules/Providers/utils/validators/provider/vsphere/validateVDDKImage';
+import { ProviderFieldsId } from 'src/providers/utils/constants';
+import type { ValidationMsg } from 'src/providers/utils/types';
+
+import type { V1beta1Provider } from '@kubev2v/types';
+
+import VMwareURLVddkEditItems from './VMwareURLVddkEditItems';
+
+type EsxiProviderEditProps = {
+  provider: V1beta1Provider;
+  onChange: (newValue: V1beta1Provider) => void;
+};
+
+const EsxiProviderEdit: FC<EsxiProviderEditProps> = ({ onChange, provider }) => {
+  const url = provider?.spec?.url;
+  const emptyVddkInitImage =
+    provider?.metadata?.annotations?.['forklift.konveyor.io/empty-vddk-init-image'];
+  const vddkInitImage = provider?.spec?.settings?.vddkInitImage;
+
+  const [urlValidation, setUrlValidation] = useState<ValidationMsg>(validateEsxiURL(url));
+  const [vddkInitImageValidation, setVddkInitImageValidation] = useState<ValidationMsg>(
+    validateVDDKImage(vddkInitImage),
+  );
+
+  const handleChange = useCallback(
+    (id: ProviderFieldsId, value: string | undefined) => {
+      const trimmedValue = value?.trim();
+
+      if (id === ProviderFieldsId.Url) {
+        setUrlValidation(validateEsxiURL(trimmedValue));
+
+        onChange({ ...provider, spec: { ...provider.spec, url: trimmedValue } });
+      }
+      if (id === ProviderFieldsId.VddkInitImage) {
+        setVddkInitImageValidation(validateVDDKImage(trimmedValue));
+
+        onChange({
+          ...provider,
+          spec: {
+            ...provider?.spec,
+            settings: {
+              ...(provider?.spec?.settings as object),
+              vddkInitImage: trimmedValue,
+            },
+          },
+        });
+      }
+      if (id === ProviderFieldsId.EmptyVddkInitImage) {
+        setVddkInitImageValidation(validateVDDKImage(undefined));
+
+        onChange({
+          ...provider,
+          metadata: {
+            ...provider?.metadata,
+            annotations: {
+              ...(provider?.metadata?.annotations as object),
+              'forklift.konveyor.io/empty-vddk-init-image': trimmedValue ?? undefined,
+            },
+          },
+          spec: {
+            ...provider?.spec,
+            settings: {
+              ...provider?.spec?.settings,
+              vddkInitImage: '',
+            },
+          },
+        });
+      }
+      if (id === ProviderFieldsId.SdkEndpoint) {
+        onChange({
+          ...provider,
+          spec: {
+            ...provider?.spec,
+            settings: {
+              ...(provider?.spec?.settings as object),
```

Comment:
u can create a function that get a path and change accordingly - or maybe we have something like that already? @avivtur